### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by DAD-FOX** [autogluon/autogluon#4803]
Why is the best L1 model I have just the KNeighbors one? Like, compared to something fancy like NeuralNetFastAI, it's super simple and should be worse, right? I don’t really get how any of this works, someone just wrote the code for me. I’m just trying to figure out what’s going on, that’s all.

TELLL ME PLESASEEEEE!

| Model                   | Score Holdout | Score Val | Eval Metric | Pred Time Test | Fit Time   |
|-------------------------|---------------|-----------|-------------|----------------|------------|
| WeightedEnsemble_L3     | 0.912756      | 0.906325  | roc_auc     | 48.835296      | -          |
| LightGBMXT_BAG_L2       | 0.912611      | 0.985697  | roc_auc     | 42.389869      | -          |
| CatBoost_BAG_L2         | 0.912564      | 0.986149  | roc_auc     | 41.597783      | -          |
| NeuralNetFastAI_BAG_L2  | 0.912524      | 0.906079  | roc_auc     | 42.694117      | -          |
| XGBoost_BAG_L2          | 0.912398      | 0.904564  | roc_auc     | 41.888102      | -          |
| LightGBMLarge_BAG_L2    | 0.912297      | 0.905416  | roc_auc     | 42.046519      | -          |
| CatBoost_n177_BAG_L2    | 0.911646      | 0.905813  | roc_auc     | 41.602232      | -          |
| RandomForestGini_BAG_L2 | 0.910671      | 0.923403  | roc_auc     | 43.378664      | -          |
| ExtraTreesEntr_BAG_L2   | 0.908066      | 0.929837  | roc_auc     | 43.515180      | -          |
| ExtraTreesGini_BAG_L2   | 0.906088      | 0.928513  | roc_auc     | 43.511286      | -          |
| WeightedEnsemble_L2     | 0.958178      | 0.955466  | roc_auc     | 0.646264       | -          |
| KNeighborsDist_BAG_L1   | 0.982879      | 0.896594  | roc_auc     | 0.602664       | -          |
| KNeighborsUnif_BAG_L1   | 0.878609      | 0.872931  | roc_auc     | 0.609228       | -          |
| LightGBM_BAG_L1         | 0.773158      | 0.738068  | roc_auc     | 0.528435       | -          |
| ExtraTreesEntr_BAG_L1   | 0.750753      | 0.728976  | roc_auc     | 2.329637       | -          |
| ExtraTreesGini_BAG_L1   | 0.742049      | 0.728797  | roc_auc     | 2.296536       | -          |
| RandomForestEntr_BAG_L1 | 0.729257      | 0.715455  | roc_auc     | 15.994952      | -          |
| LightGBMXT_BAG_L1       | 0.728783      | 0.714943  | roc_auc     | 15.990637      | -          |
| RandomForestGini_BAG_L1 | 0.682226      | 0.672343  | roc_auc     | 0.582345       | -          |
| CatBoost_BAG_L1         | 0.682226      | 0.672343  | roc_auc     | 0.528345       | -          |
| NeuralNetFastAI_BAG_L1  | 0.555044      | 0.555137  | roc_auc     | 2.499144       | -          |
| XGBoost_BAG_L1          | 0.681226      | 0.630680  | roc_auc     | 0.599998       | -          |

**Comment by DAD-FOX** [comment]
@Innixma @shchur

**Comment by DAD-FOX** [comment]
@Innixma @shchur help!!!

**Comment by DAD-FOX** [comment]
@suzhoum ?

**Comment by Innixma** [comment]
Fancy doesn't mean it is always better. Sometimes a simple model is best. Especially if your data is simple.
